framework/src/base/MooseApp.C

```diff
@@ -1477,6 +1477,43 @@ MooseApp::getMeshGeneratorNames() const
   return names;
 }
 
+std::string
+MooseApp::getFinalMeshGenerator() const
+{
+  if (_mesh_generators.empty())
+    mooseError("getFinalMeshGenerator() called but not mesh generators are present.");
+
+  DependencyResolver<std::shared_ptr<MeshGenerator>> resolver;
+
+  // Add all of the dependencies into the resolver and sort them
+  for (const auto & it : _mesh_generators)
+  {
+    // Make sure an item with no dependencies comes out too!
+    resolver.addItem(it.second);
+
+    std::vector<std::string> & generators = it.second->getDependencies();
+    for (const auto & depend_name : generators)
+    {
+      auto depend_it = _mesh_generators.find(depend_name);
+
+      if (depend_it == _mesh_generators.end())
+        mooseError("The MeshGenerator \"",
+                   depend_name,
+                   "\" was not created, did you make a "
+                   "spelling mistake or forget to include it "
+                   "in your input file?");
+
+      resolver.insertDependency(it.second, depend_it->second);
+    }
+  }
+
+  const auto & ordered_generators = resolver.getSortedValuesSets();
+  if (!ordered_generators.size())
+    mooseError("resolver.getSortedValuesSets() returned empty array.");
+
+  return ordered_generators.back().back()->name();
```

Comment:
There's already a setFinalMeshGenerator that gets called to store off information about the final generator. You should need to duplicate the execution logic to recover the final name. In fact, I'm pretty sure this isn't correct.